src/v/cluster/archival/archival_metadata_stm.cc

```diff
@@ -1323,7 +1327,7 @@ model::offset archival_metadata_stm::max_collectible_offset() {
         collect_all = false;
     }
 
-    if (collect_all || is_read_replica) {
+    if (collect_all || is_read_replica || (uploads_paused && gaps_allowed)) {
```

Comment:
what's the significance of the `uploads_paused` check here? it kinda seems like `gaps_allowed` is the flag that is doing the real work?

Reply:
we don't want to create gaps if uploads are enabled but can't keep up with data ingestion

Reply:
> we don't want to create gaps if uploads are enabled but can't keep up with data ingestion

got it. it makes a lot more sense looking at it again a second time.